packages/block-library/src/button/edit.js

```diff
@@ -183,6 +191,19 @@ function ButtonEdit( props ) {
 							: undefined,
 						...colorProps.style,
 					} }
+					onSplit={ ( value ) => {
+						if ( ! value ) {
+							return createBlock( 'core/button' );
```

Comment:
Are there any values that would be better to retain from the forked button's attributes? The style, colour and border settings come to mind. But then we'd have to be careful not to port the remaining attributes, since it  could cause unexpected and sometimes incorrect behaviour (e.g. if a hook adds HTML Anchor support to a button for some reason).

Reply:
Right, I don't really know what to do here. It sounds like generally we _should_ keep the attributes for both blocks as it makes more sense for buttons (not for paragraphs). The anchor will be an issue with this code as well since it's possible to split in the middle and end up with two buttons with the same attributes (both parts are non empty). The anchor issue should probably be fixed at another level, maybe something that makes sure the content has no duplicate anchors in general?

Reply:
I agree with keeping the attributes for both buttons.